pytorch_tabnet/tab_network.py

```diff
@@ -191,6 +179,163 @@ def forward_masks(self, x):
+
+        Parameters
+        ----------
+        - input_dim : int
+            Number of features
+        - output_dim : int or list of int for multi task classification
+            Dimension of network output
+            examples : one for regression, 2 for binary classification etc...
+        - n_d : int
+            Dimension of the prediction  layer (usually between 4 and 64)
+        - n_a : int
+            Dimension of the attention  layer (usually between 4 and 64)
+        - n_steps: int
+            Number of sucessive steps in the newtork (usually betwenn 3 and 10)
+        - gamma : float
+            Float above 1, scaling factor for attention updates (usually betwenn 1.0 to 2.0)
+        - momentum : float
+            Float value between 0 and 1 which will be used for momentum in all batch norm
+        - n_independent : int
+            Number of independent GLU layer in each GLU block (default 2)
+        - n_shared : int
+            Number of independent GLU layer in each GLU block (default 2)
+        - epsilon: float
+            Avoid log(0), this should be kept very low
+        - mask_type: str
+            Either "sparsemax" or "entmax" : this is the masking function to use
+        """
+
+        super(TabNetNoEmbeddings, self).__init__()
+        self.is_multi_task = isinstance(output_dim, list)
+        self.encoder = TabNetEncoder(input_dim,
+                                     n_d=n_d, n_a=n_a,
+                                     n_steps=n_steps, gamma=gamma,
+                                     n_independent=n_independent,
+                                     n_shared=n_shared,
+                                     epsilon=epsilon,
+                                     virtual_batch_size=virtual_batch_size,
+                                     momentum=momentum,
+                                     mask_type=mask_type)
+
+        if self.is_multi_task:
+            self.multi_task_mappings = torch.nn.ModuleList()
+            for task_dim in output_dim:
+                task_mapping = Linear(n_d, task_dim, bias=False)
+                initialize_non_glu(task_mapping, n_d, task_dim)
+                self.multi_task_mappings.append(task_mapping)
+        else:
+            self.final_mapping = Linear(n_d, output_dim, bias=False)
+            initialize_non_glu(self.final_mapping, n_d, output_dim)
+
+    def forward(self, x):
+        res, M_loss = self.encoder(x)
+
+        if self.is_multi_task:
+            # Result will be in list format
+            out = []
+            for task_mapping in self.multi_task_mappings:
+                out.append(task_mapping(res))
+        else:
+            out = self.final_mapping(res)
+
+        return out, M_loss
+
+    def forward_masks(self, x):
+        self.encoder.forward_masks(x)
+
+
+class TabNetDecoder(torch.nn.Module):
+    def __init__(self, input_dim, output_dim,
+                 n_hidden=8, # size of the hidden layer between the FT and the FC
```

Comment:
hmm, it could indeed be a good idea to give an extra parameter for the hidden dimension of the decoder